### ISSUE TITLE
Set the key to emails before they're being sent

### ISSUE THREAD

**Issue by ryssbowh** [verbb/comments#282]
### What are you trying to do?

Hi,

I created a plugin [Email](https://github.com/ryssbowh/craft-emails) which basically overrides the system messages system to add some features : Project config integration, logging, resend, cc, bcc, translations etc

Some users are reporting a conflict with the Comments plugin when it sends out emails. Comments seems to create the email with `Craft::createObject` [there](https://github.com/verbb/comments/blob/craft-3/src/services/CommentsService.php#L944) without settings its key, which is what `Mailer::composeFromKey()` would do for example.
So when the email is sent other plugins can't know what system message is being sent.

Would you be open to set the key on the emails before they're being sent ?

Original issue : https://github.com/ryssbowh/craft-emails/issues/10

Version affected : ^1.0 for sure and most likely ^2.0 according to what I see [there](https://github.com/verbb/comments/blob/craft-4/src/services/Comments.php#L1001)

### What's your proposed solution?

Use `Mailer::composeFromKey()` to create emails, or set the key to the email manually

**Comment by engram-design** [comment]
This is because we custom-render the email content, rather than rely on `Mailer::composeFromKey` which provides quick and simple mail handling, but is not as flexible. Especially if we want access to the message object, which  in our case we add Markdown support for the mail, and customer-provided templates.

I've modified [this line](https://github.com/verbb/comments/blob/cb1cbb599ef305c34581d7590b0d9e701302b2ed/src/services/CommentsService.php#L944) to be:

```php
$message = $mailer->composeFromKey($key, $variables);
```

But not seeing any change in behaviour. I've also tried:

```php
$message = Craft::createObject(['class' => $mailer->messageClass, 'mailer' => $mailer, 'key' => $key, 'variables' => $variables]);
```

Which is essentially the same thing, but the same resulting error.

Are these working on your end?

**Comment by ryssbowh** [comment]
Ha I didn't realise you were the dev there.

Thanks for that,
Yes it works for me with your second solution and Emails 1.4.7. I only tried the second solution but it would work as long as the key is set to the email object.

**Comment by engram-design** [comment]
Right, I need to re-install the plugin (either one) to get that to work, along with 1.4.8 of your plugin.

Fixed for the next release. To get this early run `composer require verbb/comments:"dev-craft-3 as 1.9.5"`

**Comment by engram-design** [comment]
Fixed in [1.9.6](https://github.com/verbb/comments/releases/tag/1.9.6)
